### PR TITLE
issues-36: Option to limit the use of machine

### PR DESCRIPTION
**Pull request by RelativoBR**
Correction of some misspelled words and inclusion of a quantity limiter of 4 MobTech machines per chuck, during the onPlace event, checking the type of material in the chunk.

### CODE REVIEW

File: src/main/java/com/github/relativobr/supreme/util/UtilMachine.java [hunk 1]
```diff
@@ -24,4 +27,22 @@ public static ItemStack getMaterial(ItemStack[] output, int randomInt) {
...
+              return true;
+            } else {
+              found++;
+            }
+          }
+        }
+      }
+    }
+    return found > limit;
+  }
```
Comment (JustAHuman-xD) [inline src/main/java/com/github/relativobr/supreme/util/UtilMachine.java:46]: I would not do this like this it's laggy and just not a great way to do it, instead I would save a counter in the chunks storage and check that instead, this also would count non -slimefun blocks and just is not something you want to do. By getting the storage in the on place and adding to it, and the opposite when removing. Just an idea though there are definitely other ways.

File: src/main/java/com/github/relativobr/supreme/machine/tech/TechRobotic.java [hunk 2]
```diff
@@ -291,4 +298,24 @@ public Radioactivity getRadioactivity() {
...
+        //check number in chuck
+        if(UtilMachine.containsLimitMaterialInChunk(block.getChunk(),
+                Supreme.getSupremeOptions().getChunkLimitMaxTechRobotic(),
+                TECH_ROBOTIC.getType())){
+          BlockStorage.clearBlockInfo(block.getLocation(), true);
+          event.setCancelled(true);
+          event.getPlayer().sendMessage(ChatColor.WHITE + "TechRobotic: " +
+                  ChatColor.RED + "Reached the machine limit for Chuck (" +
+                  Supreme.getSupremeOptions().getChunkLimitMaxTechRobotic() + "x Material)");
+        }
```
Comment (JustAHuman-xD) [inline src/main/java/com/github/relativobr/supreme/machine/tech/TechRobotic.java:317]: I would do something to help the readability of this

File: src/main/java/com/github/relativobr/supreme/machine/tech/TechMutation.java [hunk 3]
```diff
@@ -317,4 +323,24 @@ public Radioactivity getRadioactivity() {
...
+        Block block = event.getBlock();
+        //check number in chuck
+        if(UtilMachine.containsLimitMaterialInChunk(block.getChunk(),
+                Supreme.getSupremeOptions().getChunkLimitMaxTechMutation(),
+                TECH_MUTATION_I.getType())){
+          BlockStorage.clearBlockInfo(block.getLocation(), true);
+          event.setCancelled(true);
+          event.getPlayer().sendMessage(ChatColor.WHITE + "TechMutation: " +
+                  ChatColor.RED + "Reached the machine limit for Chuck (" +
+                  Supreme.getSupremeOptions().getChunkLimitMaxTechMutation() + "x Material)");
```
Comment (JustAHuman-xD) [inline src/main/java/com/github/relativobr/supreme/machine/tech/TechMutation.java:340]: Readability and potentially abstract this?

File: src/main/java/com/github/relativobr/supreme/machine/tech/TechGenerator.java [hunk 4]
```diff
@@ -435,4 +441,24 @@ public TechGenerator setSpeed(int speed) {
...
+        //check number in chuck
+        if(UtilMachine.containsLimitMaterialInChunk(block.getChunk(),
+                Supreme.getSupremeOptions().getChunkLimitMaxTechGenerator(),
+                TECH_GENERATOR.getType())){
+          BlockStorage.clearBlockInfo(block.getLocation(), true);
+          event.setCancelled(true);
+          event.getPlayer().sendMessage(ChatColor.WHITE + "TechGenerator: " +
+                  ChatColor.RED + "Reached the machine limit for Chuck (" +
+                  Supreme.getSupremeOptions().getChunkLimitMaxTechGenerator() + "x Material)");
+        }
```
Comment (JustAHuman-xD) [inline src/main/java/com/github/relativobr/supreme/machine/tech/TechGenerator.java:460]: Readability and potentially abstraction

File: src/main/java/com/github/relativobr/supreme/machine/tech/MobTechCollector.java [hunk 5]
```diff
@@ -199,4 +206,23 @@ public final MobTechCollector setMobRange(int value) {
...
+        //check number in chuck
+        if(UtilMachine.containsLimitMaterialInChunk(block.getChunk(),
+                Supreme.getSupremeOptions().getChunkLimitMaxMobTechCollector(),
+                MOB_TECH_COLLECTOR_MACHINE_I.getType())){
+          BlockStorage.clearBlockInfo(block.getLocation(), true);
+          event.setCancelled(true);
+          event.getPlayer().sendMessage(ChatColor.WHITE + "MobTechCollector: " +
+                  ChatColor.RED + "Reached the machine limit for Chuck (" +
+                  Supreme.getSupremeOptions().getChunkLimitMaxMobTechCollector() + "x Material)");
+        }
```
Comment (JustAHuman-xD) [inline src/main/java/com/github/relativobr/supreme/machine/tech/MobTechCollector.java:224]: Readability and potential abstraction

**Review by JustAHuman-xD — CHANGES_REQUESTED** [review]
Not sure that's how I would implement it, I'd look for a more performant and cleaner way.

File: src/main/java/com/github/relativobr/supreme/util/UtilMachine.java (hunk 1 above)
Comment (RelativoBR) [reply src/main/java/com/github/relativobr/supreme/util/UtilMachine.java:46]: Not sure how to implement your suggestion, could you help with that part? Or do you have any examples of these mechanisms?

**Comment by JustAHuman-xD** [comment]
And I'll look into a way to make them more performant

**Comment by RelativoBR** [comment]
new PR: https://github.com/Slimefun-Addon-Community/Supreme/pull/42

**Pull request closed without merging**